google/auth/compute_engine/credentials.py

```diff
@@ -108,3 +117,163 @@ def service_account_email(self):
     def requires_scopes(self):
         """False: Compute Engine credentials can not be scoped."""
         return False
+
+
+class Signer(crypt.Signer):
```

Comment:
We already have exactly this class here: https://github.com/GoogleCloudPlatform/google-auth-library-python/blob/master/google/auth/iam.py#L36